src/directives/text-fit/text-fit.js

```diff
@@ -14,41 +14,62 @@
 //   on the shouldWarn parameter.  This allows the caller to retry again later,
 //   perhaps after the element has become visible.
 angular.module( 'gizmos.directives' ).value( 'textFit', function textFit( element, options, shouldWarn ) {
-  var min, max, mid, lastMid, containerWidth, containerHeight
-
-  element = angular.element( element )
+  var min, max, mid, lastMid, containerStyle, containerWidth, containerHeight, projectedPercentageOfBox, accuracy, allowWordWrap
+  
+  // ToDo: get options from text-fit-group
   options = options || {}
 
-  // Min and max font size.
-  min = options.min || 6
-  max = options.max || 20
+  if ( options.debug ) console.log('[textFit] Running on: ', element.text())
 
-  containerWidth = element.parent().width()
-  containerHeight = element.parent().height()
+  // Check if element is hidden
+  if ( element[0].offsetHeight === 0 ){
+    if ( options.debug ) console.log('[textFit] hidden element: ', element.text())
+    return null
+  }
+  
+  // Set accuracy for faster guessing
+  accuracy = options.accuracy || 0
+  
+  // dis-allow word wrap
+  allowWordWrap = !(options.wordWrap === false)
+  if(!allowWordWrap) {
+    element.css('white-space', 'nowrap')
+  }
+  
+    // This is slow but WAY more reliable than el.scrollWidth. This method factors
+  // in padding and such. Slower probably not an issue since the container is 
+  // only computed once per font resize.
+  containerStyle = window.getComputedStyle(element.parent()[0])
+  if( containerStyle["box-sizing"] === 'border-box') {
+    containerWidth = parseInt(containerStyle.width, 10) - parseInt(containerStyle.paddingLeft, 10) - parseInt(containerStyle.paddingRight, 10) 
+    containerHeight = parseInt(containerStyle.height, 10)  - parseInt(containerStyle.paddingTop, 10) - parseInt(containerStyle.paddingBottom, 10)
+  } else {
+    containerWidth = parseFloat(containerStyle.width)
+    containerHeight = parseFloat(containerStyle.height)     
+  }
 
+  // Min and max font size.
+  projectedPercentageOfBox = options.projectedPercentageOfBox || .87 
+  min = options.min || 6
+  max = Math.min(containerHeight, (options.max || 120) );
+  
+  // Its assumed that initial mid should be as big as possible since most 
+  // answers will fit into regular sizes words or phrases. Size is determines by 
+  // container height devided by how many spaces used. 
+  mid = Math.floor(Math.min(containerHeight / element.text().split(' ').length, options.max || 20) * projectedPercentageOfBox * 10) / 10
```

Comment:
Yes, but upon reflection I'm going to refactor the initial guess. I noticed that it fails its purpose when height of the container can grow. I think I might just make the initial high an option since the end user will know best and too much logic is placed on the directive to figure this out defeating the purpose of making it faster.